# **Project Name**    - **Air Bnb Booking Analysis**



##### **Project Type**    - EDA
##### **Contribution**    - Individual
##### **Rakshit Pandey**

# **Project Summary -**

Write the summary here within 500-600 words.

# **GitHub Link -**

Provide your GitHub Link here.

# **Problem Statement**


**Write Problem Statement Here.**

We aim to analyze Airbnb's NYC listings to uncover patterns and insights that can enhance customer experience, optimize operational efficiency, and inform strategic business decisions.


#### **Define Your Business Objective?**

**Answer Here**

1. **Market Analysis:**

 - ***Objective:*** Understand the distribution and pricing of Airbnb listings in NYC.
 - ***Approach:*** Use data visualization to map listings by neighborhood and room type. Analyze pricing trends and identify high-demand areas and room types. This could help in strategic pricing decisions and marketing efforts.

2. **Customer Behavior Insights:**

 - ***Objective:*** Gain insights into what customers prefer and how they rate their experiences.
 - ***Approach:*** Explore correlations between listing characteristics (like room type and price) and customer reviews or ratings. Identify patterns in reviews that indicate customer satisfaction or dissatisfaction. This can help improve customer experience by addressing common pain points.

3. **Host Performance Analysis:**

 - ***Objective:*** Assess host performance and identify top performers.
 - ***Approach:*** Analyze data on the number of listings per host, the number of reviews, and average ratings. Identify characteristics of top-performing hosts, such as responsiveness or specific amenities offered, which can be shared with other hosts as best practices.

4. **Operational Optimization:**

 - ***Objective:*** Optimize listing availability and booking policies.
 - ***Approach:*** Analyze seasonal trends in availability and booking. Look at the impact of minimum stay requirements on booking rates. This can help advise hosts on adjusting availability and booking policies to maximize occupancy and revenue.

5. **Strategic Recommendations:**

 - ***Objective:*** Provide actionable insights for business growth.
 - ***Approach:*** Synthesize findings into strategic recommendations, such as potential areas for expansion, marketing focus, or new features to enhance customer satisfaction. Ensure that these recommendations are data-driven and aligned with business goals.

# **General Guidelines** : -  

1.   Well-structured, formatted, and commented code is required.
2.   Exception Handling, Production Grade Code & Deployment Ready Code will be a plus. Those students will be awarded some additional credits.
     
     The additional credits will have advantages over other students during Star Student selection.
       
             [ Note: - Deployment Ready Code is defined as, the whole .ipynb notebook should be executable in one go
                       without a single error logged. ]

3.   Each and every logic should have proper comments.
4. You may add as many number of charts you want. Make Sure for each and every chart the following format should be answered.
        

```
# Chart visualization code
```
            

*   Why did you pick the specific chart?
*   What is/are the insight(s) found from the chart?
* Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

5. You have to create at least 20 logical & meaningful charts having important insights.


[ Hints : - Do the Vizualization in  a structured way while following "UBM" Rule.

U - Univariate Analysis,

B - Bivariate Analysis (Numerical - Categorical, Numerical - Numerical, Categorical - Categorical)

M - Multivariate Analysis
 ]





# ***Let's Begin !***

## Imports

In [1]:
# installing library if not installed 

!pip install pandas numpy matplotlib seaborn --quiet

In [2]:
# Importing necessary libraries
import pandas as pd  # For data manipulation and analysis
import numpy as np  # For numerical operations
import matplotlib.pyplot as plt  # For plotting
import seaborn as sns  # For statistical data visualization

# Optional: Setting styles for plots
sns.set_style("darkgrid")


## Loading the dataset


In [3]:
# Loading the dataset

dataset = 'https://raw.githubusercontent.com/pandey-rakshit/data-library/main/Airbnb/Airbnb.csv'

# Use pandas to read the data file (CSV, Excel, etc.) into a DataFrame.

df = pd.read_csv(dataset)

## Initial Data Inspection

In [4]:
# Shape: Dimensions of the dataset - no. of rows & columns in the dataset

df.shape

(48895, 16)

In [5]:
# Head and Tail: View the first and last few rows to get a sense of the data.

df.head()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365
0,2539,Clean & quiet apt home by the park,2787,John,Brooklyn,Kensington,40.64749,-73.97237,Private room,149,1,9,2018-10-19,0.21,6,365
1,2595,Skylit Midtown Castle,2845,Jennifer,Manhattan,Midtown,40.75362,-73.98377,Entire home/apt,225,1,45,2019-05-21,0.38,2,355
2,3647,THE VILLAGE OF HARLEM....NEW YORK !,4632,Elisabeth,Manhattan,Harlem,40.80902,-73.94190,Private room,150,3,0,NaN,NaN,1,365
3,3831,Cozy Entire Floor of Brownstone,4869,LisaRoxanne,Brooklyn,Clinton Hill,40.68514,-73.95976,Entire home/apt,89,1,270,2019-07-05,4.64,1,194
4,5022,Entire Apt: Spacious Studio/Loft by central park,7192,Laura,Manhattan,East Harlem,40.79851,-73.94399,Entire home/apt,80,10,9,2018-11-19,0.10,1,0


In [6]:
df.tail()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365
48890,36484665,Charming one bedroom - newly renovated rowhouse,8232441,Sabrina,Brooklyn,Bedford-Stuyvesant,40.67853,-73.94995,Private room,70,2,0,NaN,NaN,2,9
48891,36485057,Affordable room in Bushwick/East Williamsburg,6570630,Marisol,Brooklyn,Bushwick,40.70184,-73.93317,Private room,40,4,0,NaN,NaN,2,36
48892,36485431,Sunny Studio at Historical Neighborhood,23492952,Ilgar & Aysel,Manhattan,Harlem,40.81475,-73.94867,Entire home/apt,115,10,0,NaN,NaN,1,27
48893,36485609,43rd St. Time Square-cozy single bed,30985759,Taz,Manhattan,Hell's Kitchen,40.75751,-73.99112,Shared room,55,1,0,NaN,NaN,6,2
48894,36487245,Trendy duplex in the very heart of Hell's Kitchen,68119814,Christophe,Manhattan,Hell's Kitchen,40.76404,-73.98933,Private room,90,7,0,NaN,NaN,1,23


In [7]:
# Info: Check the data types, non-null counts, and memory usage.

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

## Summary of the Dataset After First View

- ***Total Entries:*** 48,895
- ***Columns:*** 16, including both numerical (int64, float64) and categorical (object) data types.
- ***Missing Values:*** Notable in the 'name', 'host_name', 'last_review', and 'reviews_per_month' columns.
- ***Data Types:*** Most of the data types seem appropriate, but 'last_review' should be converted to a datetime format for date-specific analysis.

## Data Cleaning

### Handle Missing Values

In [8]:
# Identify missing data and decide on an approach to handle it (imputation, removal, etc.).

df.isna().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [9]:
# Impute missing values for 'name' and 'host_name'

df['name'] = df['name'].fillna('Unknown')
df['host_name'] = df['host_name'].fillna('Unknown')

In [10]:
# checking the last review & reviews_per_month have missing values in common rows

df[df['last_review'].isna().eq(True) & df['reviews_per_month'].isna().eq(True)].shape

(10052, 16)

In [11]:
# checking if the review & reviews_per_month have missing values beacuase of 0 number_of_reviews.

df[df['last_review'].isna().eq(True) & df['reviews_per_month'].isna().eq(True) & df['number_of_reviews'].eq(0)].shape

(10052, 16)

> Note: missing values of last_review & reviews_per_month are due to number_of_reviews are 0 so imputing them with ***0*** and ***no review*** respectively

In [12]:
# Impute missing values for 'last_review' and 'reviews_per_month'

df['last_review'] = df['last_review'].fillna('No Review')
df['reviews_per_month'] = df['reviews_per_month'].fillna(0)


In [13]:
# Check if there are still missing values
print(df.isna().sum())

id                                0
name                              0
host_id                           0
host_name                         0
neighbourhood_group               0
neighbourhood                     0
latitude                          0
longitude                         0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
last_review                       0
reviews_per_month                 0
calculated_host_listings_count    0
availability_365                  0
dtype: int64


### Duplicate Check

In [14]:
# Duplicate values in the dataset 

df.duplicated().sum()

0

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48895 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48895 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

## Feature Engineering

### Year Extraction & Enrichment

In [16]:
### Extracting Year from last review and saving it in new column last review year

df['last_review_year'] = pd.to_datetime(df['last_review'], errors='coerce').dropna().dt.year
df['last_review_month'] = pd.to_datetime(df['last_review'], errors='coerce').dropna().dt.month

df.sample()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,last_review_year,last_review_month
3114,1846580,Park Slope Bedroom in Eccentric New York Apart...,9646158,Frederick,Brooklyn,South Slope,40.66516,-73.9794,Private room,72,2,166,2019-07-02,2.49,1,78,2019.0,7.0


# **Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ?
Explain Briefly.

Answer Here.

# **Conclusion**

Write the conclusion here.

### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***